src/panels/lovelace/cards/hui-glance-card.ts

```diff
@@ -75,34 +82,53 @@ export class HuiGlanceCard extends hassLocalizeLitMixin(LitElement)
     const columns = config.columns || Math.min(config.entities.length, 5);
     this.style.setProperty("--glance-column-width", `${100 / columns}%`);
 
-    this.configEntities = entities;
+    this._configEntities = entities;
 
     if (this.hass) {
       this.requestUpdate();
     }
   }
 
-  protected render() {
-    if (!this.config || !this.hass) {
-      return html``;
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    if (changedProps.get("hass") && this._configEntities) {
+      for (const entity of this._configEntities) {
+        if (
+          (changedProps.get("hass") as any).states[entity.entity] !==
```

Comment:
please store the states object outside of this for-loop so you don't have to keep calling `changedProps.get` many times.